tools/update-conda.py

```diff
@@ -0,0 +1,154 @@
+                            platform_key,
+                            "--file",
+                            str(env_file),
+                            "--lockfile",
+                            str(lock_file),
+                            "--filename-template",
+                            str(lock_file),
+                        ]
+                    )
+
+
+def get_dependencies(pyproject_toml: Path) -> list[str]:
+    graystull_config = Configuration("sagemath")
+    pyproject_metadata = merge_setup_toml_metadata(
+        {}, get_all_toml_info(pyproject_toml)
+    )
+    requirements = extract_requirements(pyproject_metadata, graystull_config, {})
+    all_requirements = (
+        requirements.get("build", [])
+        + requirements.get("host", [])
+        + requirements.get("run", [])
+    )
+
+    # Specify concrete package for some virtual packages
+    all_requirements.remove("{{ blas }}")
+    all_requirements.append("blas=2.*=openblas")
+    all_requirements.remove("{{ compiler('c') }}")
+    all_requirements.append("c-compiler")
+    all_requirements.remove("{{ compiler('cxx') }}")
+    all_requirements.append("cxx-compiler")
+
+    # Correct pypi name for some packages
+    python_requirements = pyproject_metadata.get("install_requires", [])
+    # Specify concrete packages for some packages not yet in grayskull
+    # TODO: It seems to be a bug that these external.dependencies are added to install_requires by grayskull
+    python_requirements.remove("pkg:generic/tachyon")
+    python_requirements.append("tachyon")
+    python_requirements.remove("pkg:generic/sagemath-elliptic-curves")
+    python_requirements.append("sagemath-db-elliptic-curves")
+    python_requirements.remove("pkg:generic/sagemath-polytopes-db")
+    python_requirements.append("sagemath-db-polytopes")
+    python_requirements.remove("pkg:generic/sagemath-graphs")
+    python_requirements.append("sagemath-db-graphs")
+    # Following can be removed once https://github.com/regro/cf-scripts/pull/2176 is used in grayskull
+    python_requirements = [
+        req.replace("lrcalc", "python-lrcalc") for req in python_requirements
+    ]
+    all_requirements += normalize_requirements_list(
+        python_requirements, graystull_config
+    )
+    all_requirements.remove("<{ pin_compatible('numpy') }}")
+
+    # Add version constraints for some packages (not yet supported by grayskull/PEP 725)
+    all_requirements.remove("c-compiler")
+    all_requirements.append("c-compiler <=1.6")
+    all_requirements.remove("cxx-compiler")
+    all_requirements.append("cxx-compiler <=1.6")
+    return all_requirements
+
+
+def get_dev_dependencies(pyproject_toml: Path) -> list[str]:
+    pyproject = tomllib.load(pyproject_toml)
+    dependency_groups = pyproject["dependency-groups"]
+    dev_dependencies = dependency_groups.get("test", []) + dependency_groups.get(
+        "docs", []
+    )
+    return dev_dependencies
+
+
+update_conda(Path(options.sourcedir))
```

Comment:
Not super important but if we put all the global bits of this script in a `def main()` and do the usual `if __name__ == "__main__"` or similar, then this could be imported by others to use bits of the machinery for other purposes.